spec/rack/test/uploaded_file_spec.rb

```diff
@@ -25,6 +25,13 @@ def test_file_path
     expect(File.extname(uploaded_file.path)).to eq('.txt')
   end
 
+  it 'creates Tempfiles with a path that includes a single extension' do
+    uploaded_file = Rack::Test::UploadedFile.new(test_file_path)
+
+    regex = /foo#{Time.now.year}.*\.txt\Z/
+    expect(uploaded_file.path).to match(regex)
+  end
```

Comment:
For reference, this is what the test output looks like with this change, but _without_ the actual bug fix:

```
  1) Rack::Test::UploadedFile creates Tempfiles with a path that includes a single extension
     Failure/Error: expect(uploaded_file.path).to match(regex)

       expected "/var/folders/r8/mlwxnwln1vl5_t_nn90cpr5w0000gn/T/foo.txt20180327-86535-10jssch.txt" to match /foo2018.*\.txt\Z/
       Diff:
       @@ -1,2 +1,2 @@
       -/foo2018.*\.txt\Z/
       +"/var/folders/r8/mlwxnwln1vl5_t_nn90cpr5w0000gn/T/foo.txt20180327-86535-10jssch.txt"
```

...i.e. `foo.txt20180327-86535-10jssch.txt` contains the `.txt` extension _twice_.